tools/create_service_bundle.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python3
+#
+# TODO: usage, description.
+
+from datetime import date, datetime
+from typing import List
+import json
+import logging
+import os
+import re
+import retrying
+import sys
+
+from sdk_utils import groupby
+import sdk_cmd
+import sdk_diag
+import sdk_utils
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+DEFAULT_RETRY_WAIT = 1000
+DEFAULT_RETRY_MAX_ATTEMPTS = 5
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def _get_task_page(
+    offset: int, limit: int, framework_id: str = None, task_id: str = None
+) -> dict:
+    path = "/mesos/tasks?offset={}&limit={}".format(offset, limit)
+
+    if framework_id:
+        path += "&framework_id={}".format(framework_id)
+
+    if task_id:
+        path += "&task_id={}".format(task_id)
+
+    return sdk_cmd.cluster_request("GET", path).json()
+
+
+def _get_tasks(
+    tasks: dict, offset: int, limit: int, framework_id: str = None, task_id: str = None
+) -> dict:
+    page = _get_task_page(offset, limit, framework_id=framework_id, task_id=task_id)
+    page_tasks = page.get("tasks", [])
+    tasks_tasks = tasks.get("tasks", [])
+    merged_tasks = tasks_tasks + page_tasks
+    tasks["tasks"] = merged_tasks
```

Comment:
Python is pass-by-reference by default. This means that this will change the `tasks` object in the calling function. Since we pass this to the `_get_tasks` function again, it should be OK though.

Reply:
The script gets tasks through the cluster state with the latest changes.